appcues/src/main/java/com/appcues/ui/AppcuesViewModel.kt

```diff
@@ -61,7 +58,7 @@ internal class AppcuesViewModel(
 
     init {
         viewModelScope.launch {
-            stateMachine.stateFlow.collectLatest { result ->
+            experienceRenderer.stateFlow.collectLatest { result ->
```

Comment:
im not sure if i follow the change yet to hoist the stateFlow up to the experienceRenderer, rather than the state machine. This view model instance is tied to one particular experience right? so shouldn't it be listening to the state changes from a single state machine operating on that experience? What if multiple things are running at the same time, wouldn't the experienceRendering now be sending updates through this flow for all of them? even things unrelated to the experience being rendered by this view model?

Reply:
you are right!

Reply:
we need to change a couple of things here, I will see if I can make it work